cg/meta/orders/api.py

```diff
@@ -11,44 +11,36 @@
-from cg.meta.orders.fastq_submitter import FastqSubmitter
-from cg.meta.orders.fluffy_submitter import FluffySubmitter
-from cg.meta.orders.metagenome_submitter import MetagenomeSubmitter
-from cg.meta.orders.microsalt_submitter import MicrosaltSubmitter
-from cg.meta.orders.mip_dna_submitter import MipDnaSubmitter
-from cg.meta.orders.mip_rna_submitter import MipRnaSubmitter
-from cg.meta.orders.rml_submitter import RmlSubmitter
-from cg.meta.orders.rnafusion_submitter import RnafusionSubmitter
-from cg.meta.orders.sars_cov_2_submitter import SarsCov2Submitter
-from cg.meta.orders.submitter import Submitter
 from cg.meta.orders.ticket_handler import TicketHandler
-from cg.meta.orders.tomte_submitter import TomteSubmitter
 from cg.models.orders.order import OrderIn, OrderType
+from cg.services.orders.submitters.fastq_order_submitter import FastqOrderSubmitter
+from cg.services.orders.submitters.generic_order_submitter import GenericOrderSubmitter
+from cg.services.orders.submitters.metagenome_order_submitter import MetagenomeOrderSubmitter
+from cg.services.orders.submitters.microbial_order_submitter import MicrobialOrderSubmitter
+from cg.services.orders.submitters.order_submitter import OrderSubmitter
+from cg.services.orders.submitters.pool_order_submitter import PoolOrderSubmitter
 from cg.store.store import Store
 
 LOG = logging.getLogger(__name__)
 
 
-def _get_submit_handler(project: OrderType, lims: LimsAPI, status: Store) -> Submitter:
+def _get_submit_handler(project: OrderType, lims: LimsAPI, status: Store) -> OrderSubmitter:
     """Factory Method"""
 
     submitters = {
-        OrderType.BALSAMIC: BalsamicSubmitter,
-        OrderType.BALSAMIC_QC: BalsamicQCSubmitter,
-        OrderType.BALSAMIC_UMI: BalsamicUmiSubmitter,
-        OrderType.FASTQ: FastqSubmitter,
-        OrderType.FLUFFY: FluffySubmitter,
-        OrderType.METAGENOME: MetagenomeSubmitter,
-        OrderType.MICROSALT: MicrosaltSubmitter,
-        OrderType.MIP_DNA: MipDnaSubmitter,
-        OrderType.MIP_RNA: MipRnaSubmitter,
-        OrderType.RML: RmlSubmitter,
-        OrderType.RNAFUSION: RnafusionSubmitter,
-        OrderType.SARS_COV_2: SarsCov2Submitter,
-        OrderType.TOMTE: TomteSubmitter,
+        OrderType.BALSAMIC: GenericOrderSubmitter,
+        OrderType.BALSAMIC_QC: GenericOrderSubmitter,
+        OrderType.BALSAMIC_UMI: GenericOrderSubmitter,
+        OrderType.FASTQ: FastqOrderSubmitter,
+        OrderType.FLUFFY: PoolOrderSubmitter,
+        OrderType.METAGENOME: MetagenomeOrderSubmitter,
+        OrderType.MICROSALT: MicrobialOrderSubmitter,
+        OrderType.MIP_DNA: GenericOrderSubmitter,
+        OrderType.MIP_RNA: GenericOrderSubmitter,
+        OrderType.RML: PoolOrderSubmitter,
+        OrderType.RNAFUSION: GenericOrderSubmitter,
+        OrderType.SARS_COV_2: MicrobialOrderSubmitter,
+        OrderType.TOMTE: GenericOrderSubmitter,
     }
     if project in submitters:
         return submitters[project](lims=lims, status=status)
```

Comment:
I am unsure how to build these here. Since the Submitters require dependencies, they need to be built.
we can either use a factory using multiple OrderType -> service maps 
or we use a registry where we pre-instantiate the submitters. 

I think we are a bit limited here since this all happens in flask and is bit disjoint from the rest of CG as it seems